internal/handlers/sqlite/msg_update.go

```diff
@@ -108,7 +108,12 @@ func (h *Handler) updateDocument(ctx context.Context, params *common.UpdateParam
 			return 0, 0, nil, lazyerrors.Error(err)
 		}
 
-		res, err := c.Query(ctx, nil)
+		var qp backends.QueryParams
+		if !h.DisableFilterPushdown {
+			qp.Filter = u.Filter
+		}
+
+		res, err := c.Query(ctx, &qp)
```

Comment:
I guessed either this or `findAndModify` pushdown is necessary to speed up. If I'm wrong no need for this change.

> Modifies single field. We do it from Go without using any MongoDB tricks